staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -702,9 +702,12 @@ func (unstructuredJSONScheme) Encode(obj runtime.Object, w io.Writer) error {
 		for _, i := range t.Items {
 			items = append(items, i.Object)
 		}
-		t.Object["items"] = items
```

Comment:
agree this is weird, though I'm wondering if normal encoding does similar things with setting kind/apiVersion/selfLink

Reply:
It does, except in the api server we almost never encode from the original object.  Going to have to look at that.